ios/sound.mm

```diff
@@ -24,209 +24,277 @@
-  }
-
-  // special case with 4 input channels: support adding channels
-    if ( ( lNumInChan == 4 ) && bInputChMixingSupported )
+    try
     {
-        // add four mixed channels (i.e. 4 normal, 4 mixed channels)
-        lNumInChanPlusAddChan = 8;
-
-        for ( int iCh = 0; iCh < lNumInChanPlusAddChan; iCh++ )
-        {
-            int iSelCH, iSelAddCH;
-
-            GetSelCHAndAddCH ( iCh, lNumInChan, iSelCH, iSelAddCH );
-
-            if ( iSelAddCH >= 0 )
-            {
-                // for mixed channels, show both audio channel names to be mixed
-                channelInputName[iCh] =
-                    channelInputName[iSelCH] + " + " + channelInputName[iSelAddCH];
-            }
-        }
+        NSError* audioSessionError = nil;
+
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&audioSessionError];
+        [[AVAudioSession sharedInstance] requestRecordPermission:^( BOOL granted ) {
+          if ( granted )
+          {
+              // ok
+          }
+        }];
+        [[AVAudioSession sharedInstance] setMode:AVAudioSessionModeMeasurement error:&audioSessionError];
     }
-    else
+    catch ( const CGenErr& generr )
     {
-        // regular case: no mixing input channels used
-        lNumInChanPlusAddChan = lNumInChan;
+        qDebug ( "Sound exception ...." );
     }
-*/
-    // everything is ok, return empty string for "no error" case
-    return "";
+    isInitialized = false;
 }
 
-void CSound::SetLeftInputChannel  ( const int iNewChan )
+int CSound::Init ( const int iCoreAudioBufferSizeMono )
 {
-    /*// apply parameter after input parameter check
-    if ( ( iNewChan >= 0 ) && ( iNewChan < iNumInChanPlusAddChan ) )
+    try
     {
-        iSelInputLeftChannel = iNewChan;
+        // printf("Init sound ..."); <= to check the number of Sound inits at launch
+        // init base class
+        // CSoundBase::Init ( iCoreAudioBufferSizeMono ); this does nothing
+        this->iCoreAudioBufferSizeMono = iCoreAudioBufferSizeMono;
+
+        // set internal buffer size value and calculate stereo buffer size
+        iCoreAudioBufferSizeStereo = 2 * iCoreAudioBufferSizeMono;
+
+        // create memory for intermediate audio buffer
+        vecsTmpAudioSndCrdStereo.Init ( iCoreAudioBufferSizeStereo );
+
+        AVAudioSession* sessionInstance = [AVAudioSession sharedInstance];
+
+        // we are going to play and record so we pick that category
+        NSError* error = nil;
+        [sessionInstance setCategory:AVAudioSessionCategoryPlayAndRecord
+                         withOptions:AVAudioSessionCategoryOptionMixWithOthers | AVAudioSessionCategoryOptionDefaultToSpeaker |
+                                     AVAudioSessionCategoryOptionAllowBluetooth | AVAudioSessionCategoryOptionAllowBluetoothA2DP |
+                                     AVAudioSessionCategoryOptionAllowAirPlay
+                               error:&error];
```

Comment:
...
but the spacing here is wrong.  The equivalent would be `other.method ( thing, otherThing, anotherThing );`

I also don't like `error:&error` - I want more spaces... presumably after all the `:`s except with `:&` where it would be after the `:&` -- `error:& error`.

Reply:
We'll have to work that .clang-format out first

Reply:
Yeah... mostly it's just irritation having had to adapt to the spacing in the rest of Jamulus ;P

Reply:
I say let's just let those style issues slide for now. When we have a better .clang-format, it's just one command away to correct the .mm style for the whole project.